sdk/messaging/azeventgrid/autorest.md

```diff
@@ -54,11 +45,25 @@ directive:
       - response_types.go
     where: $
     transform: return $.replace(/Client(\w+)((?:Options|Response))/g, "$1$2");
+  # replace references to the "generated" CloudEvent to the actual version in azcore/messaging
   - from:
       - client.go
       - models.go
-      - models_serde.go
       - response_types.go
     where: $
-    transform: return $.replace(/AzureCoreFoundations/g, "");
+    transform: return $.replace(/\*CloudEvent/g, "messaging.CloudEvent");
+
+  # delete some models that we don't need.
+  - from:
+      - models.go
+    where: $
+    transform: return $.replace(/\/\/ (AzureCoreFoundation|CloudEvent).+?\n}/sg, "");    
```

Comment:
It might be simpler to mark the applicable types as [`x-ms-external`](https://github.com/Azure/autorest/tree/main/docs/extensions#x-ms-external) so they're omitted from codegen altogether.

Reply:
Done, this was great!